pkg/controllers/vrpq/query_reconciler.go

```diff
@@ -0,0 +1,76 @@
+/*
+ (c) Copyright [2021-2023] Open Text.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package vrpq
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	v1 "github.com/vertica/vertica-kubernetes/api/v1"
+	vapi "github.com/vertica/vertica-kubernetes/api/v1beta1"
+	"github.com/vertica/vertica-kubernetes/pkg/controllers"
+	vrpqstatus "github.com/vertica/vertica-kubernetes/pkg/vrpqstatus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+type QueryReconciler struct {
+	VRec *VerticaRestorePointsQueryReconciler
+	Vrpq *vapi.VerticaRestorePointsQuery
+	Log  logr.Logger
+}
+
+func MakeRestorePointsQueryReconciler(r *VerticaRestorePointsQueryReconciler, vrpq *vapi.VerticaRestorePointsQuery,
+	log logr.Logger) controllers.ReconcileActor {
+	return &QueryReconciler{
+		VRec: r,
+		Vrpq: vrpq,
+		Log:  log.WithName("QueryReconciler"),
+	}
+}
+
+func (q *QueryReconciler) Reconcile(ctx context.Context, req *ctrl.Request) (ctrl.Result, error) {
+	// no-op if QueryComplete is true
+	isSet := q.Vrpq.IsStatusConditionTrue(vapi.Querying)
+	if isSet {
+		return ctrl.Result{}, nil
+	}
+	return ctrl.Result{}, q.runListRestorePoints(ctx, req)
+}
+
+// setListRestorePointsQueryConditions will update the status condition before and after calling
+// list restore points api
+// Temporarily, runListRestorePoints will not call the ListRestorePoints API
+// since the dispatcher is not set up yet
+func (q *QueryReconciler) runListRestorePoints(ctx context.Context, _ *ctrl.Request) error {
+	// set Querying status condition prior to calling vclusterops API
+	err := vrpqstatus.UpdateCondition(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
+		v1.MakeCondition(vapi.Querying, metav1.ConditionTrue, ""))
+	if err != nil {
+		return err
+	}
+
+	// clear Querying status condition
+	err = vrpqstatus.UpdateCondition(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
+		v1.MakeCondition(vapi.Querying, metav1.ConditionUnknown, ""))
```

Comment:
Any reason why we didn't pick `metav1.ConditionalFalse` to turn off the status condition.  Unknown is reserved for "k8s can't decide if a resource is in the condition or not"

Reply:
yes, will update to `metav1.ConditionalFalse`